include/seqan3/search/fm_index/fm_index_cursor.hpp

```diff
@@ -106,15 +108,18 @@ class fm_index_cursor
     }
 
     //!\brief Optimized backward search without alphabet mapping
-    template <detail::sdsl_index csa_t>
-    bool backward_search(csa_t const & csa, sdsl_char_type const c, size_type & l, size_type & r) const noexcept
+    template <detail::sdsl_index sdsl_index_type>
+    bool backward_search(sdsl_index_type const & csa,
+                         sdsl_char_type const c,
+                         size_type & l,
+                         size_type & r) const noexcept
     {
         assert(l <= r && r < csa.size());
 
         size_type _l, _r;
 
         size_type cc = c;
-        if constexpr(!std::same_as<typename csa_t::alphabet_type, sdsl::plain_byte_alphabet>)
+        if constexpr(!std::same_as<typename sdsl_index_type::alphabet_type, sdsl::plain_byte_alphabet>)
```

Comment:
```suggestion
        if constexpr(!std::same_as<index_alphabet_type, sdsl::plain_byte_alphabet>)
```
we already store this too